File: 2095. Delete the Middle Node of a Linked List.cpp
class Solution {
public:
    ListNode* deleteMiddle(ListNode* headNode) {

        if (!headNode || !headNode->next) {
            return nullptr;
        }

        ListNode* slowPointer = headNode;
        ListNode* fastPointer = headNode;
        ListNode* previousNode = nullptr;

        while (fastPointer &&
               fastPointer->next) {

            previousNode = slowPointer;
            slowPointer = slowPointer->next;
            fastPointer = fastPointer->next->next;
        }

        previousNode->next = slowPointer->next;

        return headNode;
    }
};

File: 2095. Delete the Middle Node of a Linked List.js
/**
 * @param {ListNode} headNode
 * @return {ListNode}
 */
var deleteMiddle = function(headNode) {

    // If there is only one node
    if (!headNode || !headNode.next) {
        return null;
    }

    let slowPointer = headNode;

    let fastPointer = headNode;

    let previousNode = null;

    // Find the middle node
    while (fastPointer &&
           fastPointer.next) {

        previousNode =
            slowPointer;

        slowPointer =
            slowPointer.next;

        fastPointer =
            fastPointer.next.next;
    }

    // Delete the middle node
    previousNode.next =
        slowPointer.next;

    return headNode;
};
